onnxruntime/core/providers/cpu/nn/pool_base.h

```diff
@@ -183,37 +199,56 @@ class PoolBase {
                                 const int64_t kernel,
                                 int64_t* pad_head,
                                 int64_t* pad_tail,
+                                int64_t dilation,
+                                int64_t ceil_mode,
                                 int64_t* out_size) const {
     if (auto_pad_ != AutoPadType::NOTSET) {
       switch (auto_pad_) {
         case AutoPadType::VALID:
           *pad_head = 0;
           *pad_tail = 0;
-          *out_size = (in_size - kernel) / stride + 1;
+          if (ceil_mode == 0) {
+            *out_size = (in_size - dilation * (kernel - 1) - 1) / stride + 1;
+          } else {
+            *out_size = (int64_t)ceil((in_size - dilation * (kernel - 1) - 1) / (float) stride + 1);
```

Comment:
can you use c++ style cast? static_cast<int64_t>(ceil(...)) and static_cast<float>(strid)

Reply:
done